extensions/oplogPopulator/allocationStrategy/LeastFullConnector.js

```diff
@@ -6,26 +6,22 @@ const AllocationStrategy = require('./AllocationStrategy');
  * @classdesc LeastFullConnector is an allocation
  * strategy that assigns buckets to connectors based
  * on the number of buckets assigned to each connector.
- * Connectors with the fewest buckets are filled first
+ * Connectors with the fewest buckets are filled first.
+ * If a connector reached the maximum number of buckets,
+ * a new connector is created.
  */
 class LeastFullConnector extends AllocationStrategy {
-
-    /**
-     * @constructor
-     * @param {Object} params params
-     * @param {Logger} params.logger logger object
-     */
-    constructor(params) {
-        super(params.logger);
-    }
-
     /**
      * Get best connector for assigning a bucket
      * @param {Connector[]} connectors available connectors
      * @returns {Connector} connector
      */
     getConnector(connectors) {
-        return connectors.reduce((prev, elt) => (elt.bucketCount < prev.bucketCount ? elt : prev));
+        const connector = connectors.reduce((prev, elt) => (elt.bucketCount < prev.bucketCount ? elt : prev));
+        if (connector.shouldBeDestroyed || connector.buckets.length >= this._maximumBucketsPerConnector) {
+            return this._addConnector();
```

Comment:
properly handling this will probably also help avoid the `shouldBeDestroyed` connector status, which looks weird (why would the connector indicate to itself that it should be destroyed, instead of destroying itself...) : maybe that handling would be better for the `Allocator`? (same, need to look more deeply at the code)